vault/cors.go

```diff
@@ -0,0 +1,161 @@
+package vault
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+
+	"github.com/hashicorp/vault/helper/strutil"
+	"github.com/hashicorp/vault/logical"
+)
+
+var errCORSNotConfigured = errors.New("CORS is not configured")
+
+var preflightHeaders = map[string]string{
+	"Access-Control-Allow-Headers":     "*",
+	"Access-Control-Max-Age":           "1800",
```

Comment:
This is the setting we use in some of our apps here. So kind of arbitrary. This value determines how often the browser needs to make preflight requests. 

I'll see if I can find some insight around a best-practice.